### PR TITLE
Full y axis labels in Heatmap chart

### PR DESCRIPTION
**Pull request by Anush2303**
## New Behavior

Full Y Axis labels feature in Heatmap chart.

image

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: change/@fluentui-react-charting-4a2b4230-b5e3-4cea-9777-4e1670b33156.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-4a2b4230-b5e3-4cea-9777-4e1670b33156.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33509&commitId=b255427193e98bd1b5fd266257c3f602aa86bdc7&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.No callout width specified.chromium.png|2319| Changed|
|Callout.Left top edge.chromium.png|1949| Changed|
|Callout.Root.chromium.png|2309| Changed|
|Callout.Top right edge.chromium.png|1594| Changed|
|Callout.Bottom auto edge.chromium.png|2309| Changed|
|Callout.Right bottom edge.chromium.png|3005| Changed|

</details>


  <details><summary><b>react-charting-HeatMapChart</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-HeatMapChart.Basic.chromium.png|307| Changed|
|react-charting-HeatMapChart.Basic Dark Mode.chromium.png|320| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>


  <details><summary><b>react-charting-VerticalBarChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis.chromium.png|4| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx [hunk 2]
```diff
@@ -142,14 +143,22 @@ export class CartesianChartBase
   public componentDidMount(): void {
     this._fitParentContainer();
     if (
-      this.props.chartType === ChartTypes.HorizontalBarChartWithAxis &&
+      [ChartTypes.HorizontalBarChartWithAxis, ChartTypes.HeatMapChart].includes(this.props.chartType) &&
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:146]: define this array once as its being used at multiple places.

File: packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx [hunk 3]
```diff
@@ -142,14 +143,22 @@ export class CartesianChartBase
...
+        maxYAxisLabelLength = calculateLongestLabelWidth(
+          this.props.points[0].data.map((point: IHeatMapChartDataPoint) => point.y),
+          `.${this._classNames.yAxis} text`,
+        );
+      } else {
+        maxYAxisLabelLength = calculateLongestLabelWidth(
+          this.props.points.map((point: IHorizontalBarChartWithAxisDataPoint) => point.y),
+          `.${this._classNames.yAxis} text`,
+        );
+      }
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:161]: similarly this block is also repeated, can be moved to a function

File: packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx (hunk 2 above)
Comment (Anush2303) [reply packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:146]: okay.

File: packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx [hunk 4]
```diff
@@ -141,14 +143,11 @@ export class CartesianChartBase
...
-    if (
-      this.props.chartType === ChartTypes.HorizontalBarChartWithAxis &&
-      this.props.showYAxisLables &&
-      this.yAxisElement
-    ) {
-      const maxYAxisLabelLength = calculateLongestLabelWidth(
-        this.props.points.map((point: IHorizontalBarChartWithAxisDataPoint) => point.y),
-        `.${this._classNames.yAxis} text`,
+    if (chartTypesToCheck.includes(this.props.chartType) && this.props.showYAxisLables && this.yAxisElement) {
+      const maxYAxisLabelLength = this.calculateMaxYAxisLabelLength(
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:147]: why changed the function name

File: packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts [hunk 5]
```diff
@@ -108,6 +108,11 @@ export interface IHeatMapChartProps extends Pick<ICartesianChartProps, Exclude<k
    * The prop used to define the culture to localized the numbers
    */
   culture?: string;
+
+  /**
+   *@default false
+   *Used for showing complete y axis lables   */
+  showYAxisLables?: boolean;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts:115]: >showYAxisLables [](http://example.com/codeflow?start=2&length=15)

his prop is already part of cartesian. Should not be needed here.

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts:115]: yes.

File: change/@fluentui-react-charting-4a2b4230-b5e3-4cea-9777-4e1670b33156.json (hunk 1 above)
Comment (AtishayMsft) [reply change/@fluentui-react-charting-4a2b4230-b5e3-4cea-9777-4e1670b33156.json:1]: Change in VR screenshot as per changes in the PR.

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**